### PR TITLE
refactor snapshot creation and check trace data against snapshot

### PR DESCRIPTION
**Pull request by nick-gorman**
- snapshot definition is moved to the translator
- conversion to 'ns' datetime units for traces is moved to the translator
- functionality added to check traces against snapshot before they are saved by the translator
- Also changes config structure so all config inputs related to temporality sit under the 'temporal' key

### CODE REVIEW

File: src/ispypsa/translator/snapshot.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+from datetime import datetime
+
+import pandas as pd
+
+
+def create_snapshot_index(
```
Comment (prakaa) [inline src/ispypsa/translator/snapshot.py:6]: Should this be renamed to something like `create_complete_snapshot_index` or `create_full_snapshot_index` to indicate this function is creating a full, pre-aggregation set of snapshots (i.e. there might be a similar function to conduct checks post aggregation)?

File: src/ispypsa/translator/snapshot.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    """Creates a DatetimeIndex, stored in DataFrame, defining the snapshots for the model before temporal aggregation.
+
+    The index will start at the beginning of `start_year` and finish at the end of
+    `end_year` with the specified temporal resolution.
+
+    Args:
+        start_year: int specifying the start year
+        end_year: int specifying the end year
+        operational_temporal_resolution_min: int specifying the snapshot temporal resolution in minutes
+        year_type: str specifying the year type 'fy' for financial year means that start_year and end_year refer to
```
Comment (prakaa) [inline src/ispypsa/translator/snapshot.py:21]: Full stop after "year type".

**Review by prakaa — APPROVED** [review]
Look good @nick-gorman, minor renaming suggestions only

File: src/ispypsa/translator/snapshot.py (hunk 1 above)
Comment (nick-gorman) [reply src/ispypsa/translator/snapshot.py]: Good suggestion. I've changed the name to create_complete_snapshot_index.

File: src/ispypsa/translator/snapshot.py (hunk 2 above)
Comment (nick-gorman) [reply src/ispypsa/translator/snapshot.py]: Changed.

**Pull request merged**